examples/shell/shell_common/cmd_ping.cpp

```diff
@@ -0,0 +1,436 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include <core/CHIPCore.h>
+#include <messaging/ExchangeMgr.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <protocols/echo/Echo.h>
+#include <shell/shell.h>
+#include <support/CodeUtils.h>
+#include <support/ErrorStr.h>
+#include <system/SystemPacketBuffer.h>
+#include <transport/PASESession.h>
+#include <transport/SecureSessionMgr.h>
+#include <transport/raw/TCP.h>
+#include <transport/raw/UDP.h>
+
+#include <ChipShellCollection.h>
+
+using namespace chip;
+using namespace Shell;
+using namespace Logging;
+
+#define ECHO_CLIENT_PORT (CHIP_PORT + 1)
+
+constexpr size_t kMaxTcpActiveConnectionCount = 4;
+constexpr size_t kMaxTcpPendingPackets        = 4;
+constexpr size_t kNetworkSleepTimeMsecs       = (100 * 1000);
+
+namespace {
+
+constexpr Transport::AdminId gAdminId = 0;
+
+// The echo client object.
+Protocols::Echo::EchoClient gEchoClient;
+
+TransportMgr<Transport::UDP> gUDPManager;
+TransportMgr<Transport::TCP<kMaxTcpActiveConnectionCount, kMaxTcpPendingPackets>> gTCPManager;
+Messaging::ExchangeManager gExchangeManager;
+SecureSessionMgr gSessionManager;
+Inet::IPAddress gDestAddr;
+
+// Max value for the number of echo requests sent.
+uint32_t gMaxEchoCount = 3;
```

Comment:
could we group these to not be stand-alone globals and grouped by something like 'PingArguments' class?

Reply:
Acked